### PR TITLE
feat(cookies): add direct cookie access methods and update translations

### PR DESCRIPTION
**Pull request by sanish-bruno**
[Jira](https://usebruno.atlassian.net/browse/BRU-2649)

## Summary

Adds direct `bru.cookies.*` access methods that mirror Postman's `pm.cookies` API, allowing users to read, iterate, and manipulate cookies without going through `bru.cookies.jar()`.

Previously, all cookie operations required explicitly creating a jar and passing URLs:
```js
const jar = bru.cookies.jar();
const cookie = await jar.getCookie("https://example.com", "session");
```

Now, users can work with cookies scoped to the current request URL directly:
```js
const value = bru.cookies.get("session");
await bru.cookies.add({ key: "token", value: "abc" });
```

The existing `jar()` API remains fully supported and backward-compatible.

### Class hierarchy (`packages/bruno-js/src/`)

```
ReadOnlyPropertyList  (read-only base, static + dynamic modes)
  └── PropertyList    (sync mutations for static mode; throws in dynamic mode)
       └── CookieList (overrides write methods with async cookie jar delegation)
```

- **`ReadOnlyPropertyList`** — Base collection class supporting two modes:
  - *Static mode*: items stored in an internal array (for future use with headers, query params, etc.)
  - *Dynamic mode*: a `dataSource` function returns fresh items on every read (used by cookies)
- **`PropertyList`** — Adds sync mutation methods (`add`, `remove`, `upsert`, `clear`, etc.) for static mode. Throws in dynamic mode to enforce subclass overrides.
- **`CookieList`** — Cookie-specific subclass using dynamic mode. Read/iteration methods are inherited; write methods (`add`, `upsert`, `remove`, `delete`, `clear`) delegate to the cookie jar asynchronously. `jar()` provides direct jar access with URL interpolation.

### QuickJS sandbox bridge (`packages/bruno-js/src/sandbox/quickjs/utils/property-list-bridge.js`)

A reusable factory that auto-wires PropertyList methods onto QuickJS VM objects:
- **Sync read methods** — `vm.newFunction` returning marshalled values
- **Sync read object methods** — Same but with `cleanCircularJson()` for safe serialization
- **Async write methods** — `_prefix` bridge pattern: native callback → QuickJS promise
- **Iterators** — Generated eval code using `all()` as the data source

## New `bru.cookies` methods

| Category | Methods |
|---|---|
| **Read** | `get(name)`, `has(name, [value])`, `one(name)`, `all()`, `count()`, `idx(n)`, `indexOf(item)`, `toObject()`, `toString()` |
| **Iteration** | `each(fn)`, `find(fn)`, `filter(fn)`, `map(fn)`, `reduce(fn, acc)` |
| **Write** | `add(obj)`, `upsert(obj)`, `remove(name)`, `delete(name)`, `clear()` |

### Key behaviors

- **URL-scoped**: All direct methods operate on cookies matching the current request URL
- **`clear()` vs `jar().clear()`**: `bru.cookies.clear()` removes only cookies for the current URL; `jar().clear()` removes ALL cookies globally
- **`add()` = `upsert()`**: Both delegate to `jar.setCookie()` (cookie jars inherently upsert)
- **`remove()` = `delete()`**: Aliases for the same operation
- **Graceful no-URL handling**: If the request URL is unavailable, read methods return empty results and write methods resolve as no-ops

### Core implementation
- **`bru.js`** — Replaced inline cookies object with `new CookieList(...)`, accepts `requestUrl` parameter
- **`sandbox/quickjs/shims/bru.js`** — Exposed all new cookie methods to the QuickJS sandbox via `createPropertyListBridge`
- **`runtime/*.js`** (`script-runtime`, `test-runtime`, `assert-runtime`, `vars-runtime`) — Pass `request?.url` to `Bru` constructor

### Translators
- **`postman-translations.js`** — Added regex translations for `pm.cookies.*` direct access methods; reordered jar patterns before simpler patterns to prevent false matches
- **`postman-to-bruno-translator.js`** — Added AST simple translations for `pm.cookies.*`; removed complex AST transforms for `pm.cookies.has/get/toObject` (no longer needed since translations are now 1:1)
- **`bruno-to-postman-translator.js`** — Added reverse translations for `bru.cookies.*`

### Autocomplete
- **`autocomplete.js`** — Added all new `bru.cookies.*` method hints + `jar().hasCookie()`

### E2E test fixtures (`packages/bruno-tests/collection/scripting/api/bru/cookies/`)
- `directGet.bru` — Tests `get()` and `has()` (with value matching)
- `directReadMethods.bru` — Tests `one()`, `all()`, `count()`, `idx()`, `toObject()`, `toString()`
- `directIteration.bru` — Tests `each()`, `find()`, `filter()`, `map()`, `reduce()`, `indexOf()`, `idx()` edge cases
- `directWrite.bru` — Tests `add()`, `upsert()`, `remove()`, `delete()`, `clear()`
- `clearScope.bru` — Verifies `bru.cookies.clear()` only clears current URL vs `jar().clear()` clears all
- `crossDomainIsolation.bru` — Verifies cookies from other domains are not visible via `bru.cookies`
- Existing `*.bru` files updated to use `{{localhost}}` variable for consistency

### Playwright E2E tests (`tests/scripting/bru-api/cookies/`)
- `cookies.spec.ts` — Runs all cookie tests in both developer and safe sandbox modes (16 requests each)
- `runner.ts` — New `runFolder()` utility for running a specific folder within a collection

### Unit tests (`packages/bruno-js/tests/`)
- `readonly-property-list.spec.js` — Covers all read/search/iteration/transform methods, static + dynamic modes, edge cases
- `property-list.spec.js` — Covers all mutation methods in static mode, dynamic mode guards
- `cookie-list.spec.js` — Covers async write delegation, jar integration, URL interpolation, no-URL edge cases, tough-cookie normalization

## Test plan

- [x] Unit tests pass: `npm run test --workspace=packages/bruno-js`
- [x] Converter tests pass: `npm run test --workspace=packages/bruno-converters`
- [x] E2E cookie tests pass: `bru run scripting/api/bru/cookies --env=Prod` (42/42 pass)
- [ ] Manual: open desktop app, run a request with cookies, verify `bru.cookies.get()` works in pre-request/test scripts
- [ ] Manual: import a Postman collection using `pm.cookies.get()` and verify it translates correctly

#### Contribution Checklist:

- [ ] **I've used AI significantly to create this pull request**
- [ ] **The pull request only addresses one issue or adds one feature.**
- [ ] **The pull request does not introduce any breaking changes**
- [ ] **I have added screenshots or gifs to help explain the change if applicable.**
- [ ] **I have read the [contribution guidelines](https://github.com/usebruno/bruno/blob/main/contributing.md).**
- [ ] **Create an issue and link to the pull request.**

Note: Keeping the PR small and focused helps make it easier to review and merge. If you have multiple changes you want to make, please consider submitting them as separate pull requests.

#### Publishing to New Package Managers

Please see [here](../publishing.md) for more information.

## New Features

* **Enhanced Cookie Management API**: Expanded cookie operations with new methods including `get()`, `has()`, `remove()`, `clear()`, `add()`, `upsert()`, `one()`, `all()`, `count()`, `find()`, `filter()`, `each()`, `map()`, `reduce()`, and `toObject()` for more comprehensive cookie manipulation.
* **Improved Script Conversion**: Enhanced Postman-to-Bruno script translation for cookie-related operations, ensuring seamless migration.
* **IDE Autocomplete Support**: Added code completion hints for new cookie methods.

## Tests

* Comprehensive test coverage for new cookie API functionality across multiple execution modes.

### CODE REVIEW

**Comment by sanish-bruno** [comment]
@coderabbitai review

**Comment by ahamayeah59** [comment]
Glad to see a new cookies handling method. Anyway, don't forget to fix this before https://github.com/usebruno/bruno/pull/6679/changes/9d3341e46b0a69c617e832a70a6b72173e1f5c30

**Comment by sanish-bruno** [comment]
> Glad to see a new cookies handling method. Anyway, don't forget to fix this before [9d3341e](https://github.com/usebruno/bruno/commit/9d3341e46b0a69c617e832a70a6b72173e1f5c30)

sure @ahamayeah59 https://github.com/usebruno/bruno/pull/6679 the changes seems to be straight forward, will do my best to get it tested/reviewed and merged as soon as possible.

File: packages/bruno-converters/src/postman/postman-translations.js [hunk 1]
```diff
@@ -80,17 +80,39 @@ const replacements = {
...
   'pm\\.execution\\.setNextRequest\\(\'null\'\\)': 'bru.runner.stopExecution()',
-  // Direct cookie access translations (pm.cookies.has/get/toObject)
-  'pm\\.cookies\\.has\\(([^)]+)\\)': 'await bru.cookies.jar().hasCookie(req.getUrl(), $1)',
-  'pm\\.cookies\\.get\\(([^)]+)\\)': '(await bru.cookies.jar().getCookie(req.getUrl(), $1))?.value',
-  'pm\\.cookies\\.toObject\\(\\)': '(await bru.cookies.jar().getCookies(req.getUrl())).reduce((obj, c) => ({...obj, [c.key]: c.value}), {})',
-  // Cookie jar translations
+  // Cookie jar translations — order matters: these more-specific patterns must precede
+  // the simpler pm.cookies.* patterns below, since replacements are applied in insertion order.
+
   'pm\\.cookies\\.jar\\(\\)': 'bru.cookies.jar()',
```
Comment (sanish-bruno) [inline packages/bruno-converters/src/postman/postman-translations.js]: **Suggestion**: In the regex fallback path, this general `pm\.cookies\.jar\(\)` pattern runs before the more specific patterns on lines 87-91 (`pm\.cookies\.jar\(\)\.get\(`, `.set\(`, etc.). Since replacements are applied sequentially, this pattern consumes `pm.cookies.jar()` first, preventing the specific patterns from ever matching.

For example, `pm.cookies.jar().get('token')` becomes `bru.cookies.jar().get('token')` instead of the intended `bru.cookies.jar().getCookie('token')`.

This is pre-existing (the ordering was the same before this PR), and the AST translator handles it correctly, so it only affects the fallback path. But since you reorganized these patterns and added the ordering comment, it's worth fixing:

Suggested fix:
```js
// Cookie jar translations — specific method patterns MUST come before the general jar() pattern,
// otherwise jar() consumes the prefix and the method patterns never match.
'pm\\.cookies\\.jar\\(\\)\\.get\\(': 'bru.cookies.jar().getCookie(',
'pm\\.cookies\\.jar\\(\\)\\.set\\(': 'bru.cookies.jar().setCookie(',
'pm\\.cookies\\.jar\\(\\)\\.unset\\(': 'bru.cookies.jar().deleteCookie(',
'pm\\.cookies\\.jar\\(\\)\\.clear\\(': 'bru.cookies.jar().deleteCookies(',
'pm\\.cookies\\.jar\\(\\)\\.getAll\\(': 'bru.cookies.jar().getCookies(',
'pm\\.cookies\\.jar\\(\\)': 'bru.cookies.jar()',  // general pattern last
```

File: packages/bruno-js/src/readonly-property-list.js [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+  /**
+   * Get the value of an item by its key.
+   * @param {string} name
+   * @returns {*} The value property of the matching item, or undefined
+   */
+  get(name) {
+    const items = this._getItems();
+    // Use findLast so that duplicate keys resolve to the last entry,
+    // consistent with toObject() which also gives last-wins semantics.
+    const item = items.findLast((i) => i[this._keyProperty] === name);
```
Comment (sanish-bruno) [inline packages/bruno-js/src/readonly-property-list.js:64]: **Design**: `get()` uses `findLast` (last-wins) while `one()` on line 65 uses `find` (first-wins). With duplicate keys, these return inconsistent results:

```js
// items: [{key: 'x', value: 'first'}, {key: 'x', value: 'second'}]
list.get('x')          // 'second' (findLast)
list.one('x').value    // 'first'  (find)
```

The `get()` comment explains the `findLast` choice (consistent with `toObject()`), which makes sense. But a user might reasonably expect `one('x').value === get('x')`. Consider making `one()` also use `findLast` for consistency, or documenting the difference.

File: packages/bruno-app/src/utils/codemirror/autocomplete.js [hunk 3]
```diff
@@ -108,6 +108,25 @@ const STATIC_API_HINTS = {
     'bru.runner.stopExecution()',
     'bru.interpolate(str)',
     'bru.cookies',
```
Comment (sanish-bruno) [inline packages/bruno-app/src/utils/codemirror/autocomplete.js:110]: **Nit**: The actual signature is `has(name, [value])` — the optional `value` parameter lets users check both existence and value match. Consider updating to `'bru.cookies.has(name, [value])'` for discoverability.

**Comment by sanish-bruno** [comment]
@coderabbitai review

File: tests/utils/page/runner.ts [hunk 4]
```diff
@@ -79,6 +79,64 @@ export const runCollection = async (page: Page, collectionName: string) => {
...
 
+/**
+ * Runs a specific folder within a collection by navigating to it in the sidebar,
+ * opening its context menu, and clicking "Run" followed by "Recursive Run".
+ * @param page - The Playwright page object
+ * @param collectionName - The name of the collection containing the folder
+ * @param folderPath - Array of folder names forming the path (e.g. ['scripting', 'api', 'bru', 'cookies'])
+ */
+export const runFolder = async (page: Page, collectionName: string, folderPath: string[]) => {
+  await test.step(`Run folder "${folderPath.join('/')}" in "${collectionName}"`, async () => {
```
[the remark this answers is not in the document]
Comment (sanish-bruno) [reply tests/utils/page/runner.ts:90]: Not needed. This is an internal test utility, not a public API. If someone passes an empty array, the for loop is a no-op and the subsequent folderPath[folderPath.length - 1] returns undefined, which would immediately fail with a clear Playwright selector error. Adding a guard for an impossible misuse of an internal helper is over-engineering.

**Comment by sanish-bruno** [comment]
@coderabbitai review
